### ISSUE TITLE
Accessibility of select lists

### ISSUE THREAD

**Issue by btopro** [Dogfalo/materialize#3490]
When applying the select style to select lists that have labels it breaks the convention of having the label right next to the item its tied to. It goes from
```
<label for="stuff"></label>
<select id="staff">....</select>
```
to this
```
<label for="stuff"></label>
<input .... />
<select id="staff">....</select>
```
The input field should have an ID so that there's a label for it. Possible way to do this:
- generating a hashed id for the input field
- changing the for attribute on the label to the hash

The select is hidden so I think this would resolve the accessibility flag seen below.

![2016-08-10_08-32-05](https://cloud.githubusercontent.com/assets/329735/17553649/f700cb30-5ed4-11e6-9703-4b07b7e0d052.png)

**Comment by btopro** [comment]
in testing, the label swap fixes the issue but generates a new one as there is no label associated with the original select. This most likely doesn't matter because the original select field is hidden and hopefully would be skipped by screen readers.

**Comment by btopro** [comment]
https://github.com/Dogfalo/materialize/pull/3526 pull request to match this
